roles/kubernetes/master/tasks/encrypt-at-rest.yml

```diff
@@ -1,4 +1,25 @@
 ---
+- name: Check if secret for encrypting data at rest already exist
+  stat:
+    path: "{{ kube_cert_dir }}/secrets_encryption.yaml"
+  register: secrets_encryption_file
+
+- name: Slurp secrets_encryption file if it exists
+  slurp:
+    src: "{{ kube_cert_dir }}/secrets_encryption.yaml"
+  register: secret_file_encoded
+  when: secrets_encryption_file.stat.exists
+
+- name: Base 64 Decode slurped secrets_encryption file
+  set_fact:
+    secret_file_decoded: "{{secret_file_encoded['content'] | b64decode | from_yaml}}"
+  when: secrets_encryption_file.stat.exists
+
+- name: Query for secret value
+  set_fact:
+    kube_encrypt_token: "{{ secret_file_decoded | json_query(secrets_encryption_query) | first | b64decode}}"
+  when: secrets_encryption_file.stat.exists
+
```

Comment:
The potential issue is with consistency of the secret across masters.  This code is really still dependent on having the secret stored in a file on a workstation (which causes issues when multiple individuals are running deployments and the secret can't be checked into source).  Consider the case when a different cluster administrator adds a new master causing a new secret to be generated.  If we simplify things a bit and assume each existing master has an encryption secret properly configured, then the secret could be pulled off any master that has a secrets_encryption.yaml and copied over.

Reply:
I didn't consider a situation where the a master would be missing the encryption secret and it seems like a valid concern. I added an additional task to make sure that the secret gets set on all nodes in kube-master.